src/estimagic/optimization/tranquilo/region.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass, replace
+
+import numpy as np
+
+from estimagic.optimization.tranquilo.bounds import Bounds
+from estimagic.optimization.tranquilo.volume import (
+    get_radius_of_cube_with_volume_of_sphere,
+)
+
+
+@dataclass
+class Region:
+    """Trust region."""
+
+    center: np.ndarray
+    radius: float
+    bounds: Bounds = None
+
+    @property
+    def shape(self) -> str:
+        any_bounds_binding = _any_bounds_binding(
+            bounds=self.bounds, center=self.center, radius=self.radius
+        )
+        return "cube" if any_bounds_binding else "sphere"
+
+    @property
+    def cube_bounds(self) -> Bounds:
+        if self.shape == "sphere":
+            radius = self.radius
```

Comment:
I think we should do one of the two if `shape == "sphere"`
1. raise an Attribute error for cube_bounds
2. return cube_bounds but also use a cube_radius

Simply returning bounds constructed from a sphere radius is not a good option in my opinion. I am tending to option 2.